subprojects/core/src/main/java/org/gradle/internal/progress/TaskOutcomeStatisticsFormatter.java

```diff
@@ -19,23 +19,38 @@
 import org.gradle.api.internal.tasks.TaskStateInternal;
 import org.gradle.api.tasks.TaskState;
 
+import java.util.concurrent.atomic.AtomicInteger;
+
 public class TaskOutcomeStatisticsFormatter {
-    private int executedTasksCount;
-    private int allTasksCount;
+    private AtomicInteger avoidedTasksCount = new AtomicInteger(0);
+    private AtomicInteger allTasksCount = new AtomicInteger(0);
 
-    public String incrementAndGetProgress(TaskState state) {
+    public String incrementAndGetProgress(final TaskState state) {
         recordTaskOutcome(state);
 
-        final long executedPercentage = Math.round(executedTasksCount * 100.0 / allTasksCount);
-        return " [" + (100 - executedPercentage) + "% AVOIDED, " + executedPercentage + "% DONE]";
+        if (allTasksCount.get() > 0) {
+            final long avoidedPercentage = Math.round(avoidedTasksCount.get() * 100.0 / allTasksCount.get());
+            return " [" + avoidedPercentage + "% AVOIDED, " + (100 - avoidedPercentage) + "% DONE]";
+        } else {
+            return "";
+        }
     }
 
     private void recordTaskOutcome(final TaskState state) {
         TaskStateInternal stateInternal = (TaskStateInternal) state;
-        TaskExecutionOutcome outcome = stateInternal.getOutcome();
-        if (outcome == TaskExecutionOutcome.EXECUTED) {
-            executedTasksCount++;
+
+        if (shouldCountTask(stateInternal)) {
+            allTasksCount.getAndIncrement();
+            if (stateInternal.getUpToDate() || stateInternal.isFromCache()) {
```

Comment:
getUpToDate will be true any time isFromCache will be true because `FROM_CACHE` is a form of `UP-TO-DATE`.